src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../react-app-env';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
+
+export const PostsList: React.FC<Props> = ({
+  userSelectedId,
+  selectPost,
+  post,
+}) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
+
+  const allPosts = async () => {
+    const result = await getAllPosts();
+
+    setPostList(result);
+  };
+
+  const findposts = async () => {
+    const result = await getPosts(userSelectedId);
+
+    setPostList(result);
+  };
+
+  useEffect(() => {
+    findposts();
+  }, [userSelectedId]);
```

Comment:
also no need in separate findPost method:
```suggestion
  useEffect(() => {
    getPosts(userSelectedId).then(setPostList);
  }, [userSelectedId]);
```